store/store.go

```diff
@@ -317,46 +327,24 @@ func init() {
 		panic(err)
 	}
 
-	defaultConfig.SearchURI, err = storeBaseURI.Parse("api/v1/snaps/search")
-	if err != nil {
-		panic(err)
-	}
-
-	// slash at the end because snap name is appended to this with .Parse(snapName)
-	defaultConfig.DetailsURI, err = storeBaseURI.Parse("api/v1/snaps/details/")
-	if err != nil {
-		panic(err)
-	}
-
-	defaultConfig.BulkURI, err = storeBaseURI.Parse("api/v1/snaps/metadata")
+	assertsBaseURI, err := url.Parse(assertsURL(storeBaseURI))
 	if err != nil {
 		panic(err)
 	}
 
-	assertsBaseURI, err := url.Parse(assertsURL())
+	myappsBaseURI, err := url.Parse(myappsURL())
 	if err != nil {
 		panic(err)
 	}
 
-	defaultConfig.AssertionsURI, err = assertsBaseURI.Parse("assertions/")
-	if err != nil {
-		panic(err)
-	}
-
-	defaultConfig.OrdersURI, err = url.Parse(myappsURL() + "purchases/v1/orders")
-	if err != nil {
-		panic(err)
-	}
-
-	defaultConfig.CustomersMeURI, err = url.Parse(myappsURL() + "purchases/v1/customers/me")
-	if err != nil {
-		panic(err)
-	}
-
-	defaultConfig.SectionsURI, err = storeBaseURI.Parse("api/v1/snaps/sections")
-	if err != nil {
-		panic(err)
-	}
+	defaultConfig.SearchURI = urlJoin(storeBaseURI, "api/v1/snaps/search")
+	// slash at the end because snap name is appended to this with .Parse(snapName)
+	defaultConfig.DetailsURI = urlJoin(storeBaseURI, "api/v1/snaps/details/")
+	defaultConfig.BulkURI = urlJoin(storeBaseURI, "api/v1/snaps/metadata")
+	defaultConfig.AssertionsURI = urlJoin(assertsBaseURI, "assertions/")
```

Comment:
I thought about moving `assertions/` up, but I don't like that because it's not actually the base of the service: it would be very confusing if for some reason we needed to use `/v1/public-key-info`, for instance.  I've added a bit more commentary to try to explain what's going on here.

I deliberately didn't push `api/` into `storeBaseURI` because, as Bret observed, the `api/` prefix is redundant now that the primary hostname is `api.snapcraft.io`, so we'll probably drop it once we have redesigned v2 endpoints.  When that happens, I can easily imagine us wanting to switch one endpoint at a time, and that would be difficult if the base URI ended with `api/`.  I've added another comment to explain this.